File: packages/ios/Sources/Components/PasswordInput/PasswordInput.figma.swift
import SwiftUI
import Figma

struct KozmosPasswordInputConnect: FigmaConnect {
    let component = KozmosPasswordInput.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=412-2642"

    @FigmaString("Label Text")
    var label: String = "Password"

    @FigmaString("Placeholder Text")
    var placeholder: String = "Password"

    @FigmaString("Helper Text")
    var helperText: String = "Use at least 8 characters."

    @FigmaBoolean("Show Helper Text")
    var showHelperText: Bool = true

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Focus": "focus",
            "Disabled": "disabled",
            "Readonly": "readonly"
        ]
    )
    var state: String = "default"

    @FigmaEnum(
        "Status",
        mapping: [
            "Default": KozmosInputStatus.`default`,
            "Error": KozmosInputStatus.error,
            "Warning": KozmosInputStatus.warning,
            "Success": KozmosInputStatus.success
        ]
    )
    var status: KozmosInputStatus = .default

    @FigmaEnum(
        "Visibility",
        mapping: [
            "Hidden": false,
            "Visible": true
        ]
    )
    var defaultVisible: Bool = false

    var body: some View {
        KozmosPasswordInput(
            text: .constant(""),
            label: self.label,
            placeholder: self.placeholder,
            disabled: self.state == "disabled",
            readOnly: self.state == "readonly",
            status: self.status,
            helperText: self.showHelperText ? self.helperText : nil,
            defaultVisible: self.defaultVisible
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/PasswordInput/PasswordInput.figma.kt
package com.kozmos.components.passwordinput

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.components.input.KozmosInputStatus

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=412-2642")
class KozmosPasswordInputConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Password"

    @FigmaProperty(FigmaType.Text, "Placeholder Text")
    val placeholder: String = "Password"

    @FigmaProperty(FigmaType.Text, "Helper Text")
    val helperText: String = "Use at least 8 characters."

    @FigmaProperty(FigmaType.Boolean, "Show Helper Text")
    val showHelperText: Boolean = true

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Focus" to "focus",
        "Disabled" to "disabled",
        "Readonly" to "readonly"
    )

    @FigmaProperty(FigmaType.Enum, "Status")
    val status: KozmosInputStatus = Figma.mapping(
        "Default" to KozmosInputStatus.Default,
        "Error" to KozmosInputStatus.Error,
        "Warning" to KozmosInputStatus.Warning,
        "Success" to KozmosInputStatus.Success
    )

    @FigmaProperty(FigmaType.Enum, "Visibility")
    val defaultVisible: Boolean = Figma.mapping(
        "Hidden" to false,
        "Visible" to true
    )

    @Composable
    fun ComponentExample() {
        KozmosPasswordInput(
            value = "",
            onValueChange = {},
            label = label,
            placeholder = placeholder,
            enabled = state != "disabled",
            readOnly = state == "readonly",
            status = status,
            helperText = if (showHelperText) helperText else null,
            defaultVisible = defaultVisible
        )
    }
}
